Sources/SwiftBasicFormat/BasicFormat.swift

```diff
@@ -362,6 +393,12 @@ open class BasicFormat: SyntaxRewriter {
       anchorPoints[token] = currentIndentationLevel
     }
 
+    if requiresTrailingNewline(token) {
+      if !trailingTrivia.endsWithNewline && nextToken?.leadingTrivia.startsWithNewline == false {
```

Comment:
Shouldn’t you be checking for `nextTokenWillStartWithNewline` here to make sure we’re not adding both a trailing newline here and a leading newline to the next token?
```suggestion
      if !trailingTrivia.endsWithNewline && !nextTokenWillStartsWithNewline {
```